acquire/outputs/zip.py

```diff
@@ -0,0 +1,100 @@
+import io
+import pyzipper
+from pathlib import Path
+from typing import BinaryIO, Optional, Union
+from datetime import datetime
+
+from dissect.target.filesystem import FilesystemEntry
+
+from acquire.crypt import EncryptedStream
+from acquire.outputs.base import Output
+
+
+class ZipOutput(Output):
+    """zip archive acquire output format. Output can be compressed and/or encrypted.
+
+    Args:
+        path: The path to write the zip archive to.
+        compress: Whether to compress the zip archive.
+        encrypt: Whether to encrypt the zip archive.
+        public_key: The RSA public key to encrypt the header with.
+    """
+
+    def __init__(
+        self,
+        path: Path,
+        compress: bool = False,
+        encrypt: bool = False,
+        public_key: Optional[bytes] = None,
+    ) -> None:
+        ext = ".zip" if ".zip" not in path.suffixes else ""
+
+        self._fh = None
+        self.path = path.with_suffix(path.suffix + ext)
+
+        if compress:
+            self.compression = pyzipper.ZIP_LZMA
+        else:
+            self.compression = pyzipper.ZIP_STORED
+
+        if encrypt:
```

Comment:
It's preferred to use `EncryptedStream` instead of the encryption of ZIP itself, so that all crypto operations are universal on all acquire output files.

Example:
https://github.com/fox-it/acquire/blob/main/acquire/outputs/tar.py#L43